node/network/availability-distribution/src/lib.rs

```diff
@@ -384,27 +378,16 @@ where
 	for added in view.difference(&old_view) {
 		let validators = query_validators(ctx, *added).await?;
 		let validator_index = obtain_our_validator_index(&validators, keystore.clone()).await;
-		state
-			.add_relay_parent(ctx, *added, validators, validator_index)
-			.await?;
+		state.add_relay_parent(ctx, *added, validators, validator_index).await?;
 	}
 
 	// handle all candidates
 	for (candidate_hash, _receipt) in state.cached_live_candidates_unioned(view.difference(&old_view)) {
-		let per_candidate = state.per_candidate.entry(candidate_hash).or_default();
-
-		// assure the node has the validator role
-		if per_candidate.validator_index.is_none() {
```

Comment:
This check is no longer done, that I can see. Rationale for removal?

Reply:
I removed this by intention. 

I think that is wrong, however that means I also need to change more in here.

Let me explain what I think:

I removed it because below we are also checking the message vault. Non validator nodes also hold this message vault and will be able to use this to propagate chunks. This was the reason I removed this check.

However I think, I'm wrong here. If we switch to a new view, we can only send new chunks to peers for that we have the availability data in the store and that just got "activated" by this view. Every other candidate that was already known by us, should already be exchanges with the peers as far as we know their view. So, continuing on this, we don't need to check the message vault, as there shouldn't be any messages in it yet, but I will bring this check back.

Reply:
I don't think we really need this check. Ultimately, we don't distribute pieces unless we have them in our availability store. As a non-validator, or indeed as a non backing validator for this particular candidate, we won't have those pieces.

Reply:
I have brought it back and added some further optimizations.